x/stakeibc/keeper/unbonding_records_get_host_zone_unbondings_msgs_test.go

```diff
@@ -1,226 +1,813 @@
 
-	// TODO: check epoch unbonding record ids here
-	msgs, totalAmtToUnbond, _, _, err := s.App.StakeibcKeeper.GetHostZoneUnbondingMsgs(s.Ctx, tc.hostZone)
-	// error should be nil -- we do NOT raise an error on a non-existent chain id, we simply do not send any messages
-	s.Require().Nil(err, "error should be nil -- we do NOT raise an error when no records exist, we simply do not send any messages")
-	// no messages should be sent
-	s.Require().Equal(0, len(msgs), "no messages should be sent")
-	// no value should be unbonded
-	s.Require().Equal(sdkmath.ZeroInt(), totalAmtToUnbond, "no value should be unbonded")
+	// To make the error easier to understand, we first compare just the list of validator addresses
+	actualValidators := []string{}
+	for _, actual := range actualSortedCapacities {
+		actualValidators = append(actualValidators, actual.ValidatorAddress)
+	}
+	expectedValidators := []string{}
+	for _, expected := range expectedSortedCapacities {
+		expectedValidators = append(expectedValidators, expected.ValidatorAddress)
+	}
+	s.Require().Equal(expectedValidators, actualValidators, "validator order")
+
+	// Then we'll do a sanity check on each field
+	// If the above passes and this fails, that likely means the test was setup improperly
+	for i, expected := range expectedSortedCapacities {
+		actual := actualSortedCapacities[i]
+		address := expected.ValidatorAddress
+		s.Require().Equal(expected.ValidatorAddress, actual.ValidatorAddress, "validator %d address", i+1)
+		s.Require().Equal(expected.BalancedDelegation, actual.BalancedDelegation, "validator %s balanced", address)
+		s.Require().Equal(expected.CurrentDelegation, actual.CurrentDelegation, "validator %s current", address)
+		s.Require().Equal(expected.Capacity, actual.Capacity, "validator %s capacity", address)
+	}
 }
 
-func (s *KeeperTestSuite) TestGetHostZoneUnbondingMsgs_UnbondingTooMuch() {
-	tc := s.SetupGetHostZoneUnbondingMsgs()
+func (s *KeeperTestSuite) TestGetUnbondingICAMessages() {
+	delegationAddress := "cosmos_DELEGATION"
 
-	// iterate the validators and set all their delegated amounts to 0
-	for i := range tc.hostZone.Validators {
-		tc.hostZone.Validators[i].Delegation = sdkmath.ZeroInt()
+	hostZone := types.HostZone{
+		ChainId:              HostChainId,
+		HostDenom:            Atom,
+		DelegationIcaAddress: delegationAddress,
 	}
-	// write the host zone with zero-delegation validators back to the store
-	s.App.StakeibcKeeper.SetHostZone(s.Ctx, tc.hostZone)
-
-	// TODO: check epoch unbonding record ids here
-	_, _, _, _, err := s.App.StakeibcKeeper.GetHostZoneUnbondingMsgs(s.Ctx, tc.hostZone)
-	s.Require().EqualError(err, fmt.Sprintf("Could not unbond %v on Host Zone %s, unable to balance the unbond amount across validators: not found", tc.amtToUnbond.Mul(sdkmath.NewInt(int64(len(tc.epochUnbondingRecords)))), tc.hostZone.ChainId))
-}
 
-func (s *KeeperTestSuite) TestGetTargetValAmtsForHostZone_Success() {
```

Comment:
why did we remove the TestGetTargetValAmtsForHostZone_Success test?

Reply:
It's better covered by [this](https://github.com/Stride-Labs/stride/blob/lsm/x/stakeibc/keeper/validator_selection_test.go#L180) test that was added during the rebalance PR